### PR TITLE
[3.4] Add Phpunit 6 compatibility 

### PR DESCRIPTION
**Pull request by antograssiot**
Since CakePHP 3.4 required php5.6 as minimal required version, it implicitly allows to only support PHPUnit 5+.

Adding those couple of aliases will allow CakePHP users working on php7+ to pin to the latest version so I though that it could be a nice addition to 3.5.

I'll provide a PR to the documentation/migration guide if those modifications are accepted.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -56,7 +56,10 @@
             "Company\\TestPluginThree\\": "tests/test_app/Plugin/Company/TestPluginThree/src",
             "Company\\TestPluginThree\\Test\\": "tests/test_app/Plugin/Company/TestPluginThree/tests",
             "PluginJs\\": "tests/test_app/Plugin/PluginJs/src"
-        }
+        },
+        "files": [
+          "src/TestSuite/phpunit-forward-compat.php"
```
Comment (ADmad) [inline composer.json]: Indentation is off.

**Comment by lorenzo** [comment]
Why not target the master branch?

**Comment by antograssiot** [comment]
@ADmad indentation fixed.

@lorenzo There are already a bunch of changes and didn't wanted to push this just before the stable release but it would work of course, so I'm not against changing the branch if you guys think it's worth it.

File: src/TestSuite/phpunit-forward-compat.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+if (class_exists('PHPUnit_Runner_Version')) {
+    class_alias('PHPUnit_Framework_Test', 'PHPUnit\Framework\Test');
+    class_alias('PHPUnit_Framework_TestSuite', 'PHPUnit\Framework\TestSuite');
+    class_alias('PHPUnit_Exception', 'PHPUnit\Exception');
+    class_alias('PHPUnit_Framework_Error_Warning', 'PHPUnit\Framework\Error\Warning');
+    class_alias('PHPUnit_Framework_Constraint', 'PHPUnit\Framework\Constraint\Constraint');
+    class_alias('PHPUnit_Framework_AssertionFailedError', 'PHPUnit\Framework\AssertionFailedError');
```
Comment (markstory) [inline src/TestSuite/phpunit-forward-compat.php]: Neat way to solve this :+1:

**Review by markstory — APPROVED** [review]
Looks good :+1:

**Comment by lorenzo** [comment]
I think this can target master :)

**Comment by ADmad** [comment]
@antograssiot I took the liberty of re-targeting this to master.

**Comment by antograssiot** [comment]
Ok cool. I'll send another PR later to fix risky tests. PHPUnit is now more strict now.

**Comment by dereuromark** [comment]
I get an error when using latest dev master and PHP 5.6:

> Fatal error: Declaration of Cake\TestSuite\Fixture\FixtureInjector::addError(PHP
Unit\Framework\Test $test, Exception $e, $time) must be compatible with PHPUnit_
Framework_TestListener::addError(PHPUnit_Framework_Test $test, Exception $e, $ti
me) in ...\vendor\cakephp\cakephp\src\T
estSuite\Fixture\FixtureInjector.php on line 29

Downloaded a fresh phpunit-5.7.9.phar

**Comment by antograssiot** [comment]
I didn't tried with a phar but I'might pretty sure that the forward compatible classes provided by PHPUnit A are not present.
https://github.com/sebastianbergmann/phpunit/tree/5.7/src/ForwardCompatibility
Maybe we should check for existence and alias those also

**Comment by dereuromark** [comment]
I get the same with phpunit in composer json and vendor/bin/phpunit by the way.
Latest dev-master of 3.4.

**Comment by antograssiot** [comment]
I've open #10254 to fix it

**Pull request merged**